### PR TITLE
Swap functional test

### PR DESCRIPTION
**Pull request by TheCharlatan**
Before running the test, the docker-containers have to be started again. in /tests do `docker-compose up`, then `sudo chown -R $USER data_dir`.

I currently run the test with:
```
cargo test -- --nocapture --ignored swap_test
```

The processes don't exit cleanly yet, for cleanup I run `rm -rf tests/.farcaster_node_0 tests/.farcaster_node_1  && sudo killall peerd swapd farcasterd walletd syncerd` from the project's root directory.

### CODE REVIEW

**Comment by TheCharlatan** [comment]
I marked this ready for review, but it seems like we again have issues with the containers. This also includes hacky work to get the Local network type accepted, which conflicts with the work in #230. This should not be merged before that.

File: Cargo.toml [hunk 1]
```diff
@@ -92,6 +92,8 @@ tokio = { version = "1.12.0", features = ["full"] }
 toml = { version = "0.5", optional = true }
 # IPC
 zmq = "0.9"
+# Test
+strip-ansi-escapes = "0.1.1"
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply Cargo.toml:96]: Oh, right!

**Review by h4sh3d — COMMENTED** [review]
Some general comments:
You should be able to remove mainnet and testnet in config files, as we don’t care in this context, config will be None for those.

You can add this to run in ci tests I think.

File: Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply Cargo.toml]: Done.

File: tests/docker-compose.yml [hunk 2]
```diff
@@ -45,4 +45,10 @@ services:
             - "monerod"
         ports:
             - 18084:18083
+    monero-wallet-rpc-3:
+        image: ghcr.io/farcaster-project/containers/monero-wallet-rpc:0.17.2.3
+        environment:
+            MONEROD_ADDRESS: monerod:18081
+        ports:
```
Comment (h4sh3d) [inline tests/docker-compose.yml:52]: We can also use `depends_on` here, it just tell docker if monerod cannot start don't start the rest, but it doesn't wait for monerod to be "ready", so this is just a minor thing but we have it elsewhere

**Review by h4sh3d — APPROVED** [review]
I can merge after the tweak in docker-compose file

File: tests/docker-compose.yml (hunk 2 above)
Comment (TheCharlatan) [reply tests/docker-compose.yml:54]: Done in 9ed75c7563406834645aceac8a064c33b37fc9ee

**Pull request merged**